iceoryx_dust/test/moduletests/test_container_fixed_position_container.cpp

```diff
@@ -92,6 +94,326 @@ TEST_F(FixedPositionContainer_test, Capacity)
+
+    EXPECT_THAT(copy_sut_complex.size(), Eq(EXPECTED_SIZE));
+
+    // actually call copyAssignment
+    EXPECT_THAT(copy_sut_complex.begin()->stats.copyAssignment, Eq(EXPECTED_SIZE));
+    EXPECT_THAT(copy_sut_complex.begin()->value, Eq(EXPECTED_VALUE));
+}
+
+TEST_F(FixedPositionContainer_test, UsingCopyCtorMultipleElementsContainerPreservesAllElements)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "6261f53e-8089-4b9b-9b2d-9da0016a2f1e");
+
+    constexpr SutComplex::IndexType EXPECTED_SIZE{4U};
+    std::vector<DataType> EXPECTED_VALUE = {0U, 1U, 2U, 3U};
+    for (SutComplex::IndexType i = 0; i < EXPECTED_SIZE; ++i)
+    {
+        sut_complex.emplace(EXPECTED_VALUE[i]);
+    }
+
+    SutComplex copy_sut_complex{sut_complex};
+
+    EXPECT_THAT(copy_sut_complex.size(), Eq(EXPECTED_SIZE));
+
+    // actually call copyAssignment
+    EXPECT_THAT(copy_sut_complex.begin()->stats.copyAssignment, Eq(EXPECTED_SIZE));
+    for (Sut::IndexType i = 0; i < EXPECTED_SIZE; ++i)
+    {
+        EXPECT_THAT(copy_sut_complex.iter_from_index(i)->value, Eq(EXPECTED_VALUE[i]));
+    }
+}
+
+TEST_F(FixedPositionContainer_test, UsingCopyCtorFullCapacityContainerPreservesAllElements)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "028704df-b2f3-4133-9c16-b9d2c6a79916");
+
+    fillSutComplex();
+    constexpr SutComplex::IndexType EXPECTED_SIZE{CAPACITY};
+
+    SutComplex copy_sut_complex{sut_complex};
+
+    EXPECT_THAT(copy_sut_complex.full(), Eq(true));
+
+    // actually call copyAssignment
+    EXPECT_THAT(copy_sut_complex.begin()->stats.copyAssignment, Eq(EXPECTED_SIZE));
+    for (Sut::IndexType i = 0; i < EXPECTED_SIZE; ++i)
+    {
+        EXPECT_THAT(copy_sut_complex.iter_from_index(i)->value, Eq(sut_complex.iter_from_index(i)->value));
+        EXPECT_THAT(copy_sut_complex.iter_from_index(i), Ne(sut_complex.iter_from_index(i)));
+    }
+}
+
+// END test copy constructor
+
+// BEGIN test move constructor
+
+TEST_F(FixedPositionContainer_test, UsingMoveCtorFromEmptyContainerResultsInEmptyContainer)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "af8958fb-9a09-4987-b290-ce41abdc2354");
+
+    SutComplex move_sut_complex{std::move(sut_complex)};
+
+    EXPECT_THAT(move_sut_complex.empty(), Eq(true));
+}
+
+TEST_F(FixedPositionContainer_test, UsingMoveCtorFromSingleElementContainerClearsOriginal)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "df6c1884-43c6-4d1e-b889-6cbf4b9ee726");
+
+    constexpr DataType EXPECTED_VALUE{42U};
+    constexpr SutComplex::IndexType EXPECTED_SIZE{1U};
+    sut_complex.emplace(EXPECTED_VALUE);
+
+    SutComplex move_sut_complex{std::move(sut_complex)};
+
+    ASSERT_THAT(move_sut_complex.size(), Eq(EXPECTED_SIZE));
+
+    // actually call moveAssignment
+    EXPECT_THAT(move_sut_complex.begin()->stats.moveAssignment, Eq(EXPECTED_SIZE));
+
+    // make sure clear() been called
+    EXPECT_THAT(move_sut_complex.begin()->stats.dTor, Eq(EXPECTED_SIZE));
+    EXPECT_THAT(sut_complex.empty(), Eq(true));
```

Comment:
Please also check `size`, `empty` and `full`. It is sufficient to do it only once for `sut_complex`